test/e2e/clusterinfo/openshift.go

```diff
@@ -0,0 +1,76 @@
+package clusterinfo
+
+import (
+	"context"
+	"fmt"
+
+	v1 "github.com/openshift/api/config/v1"
+	clientset "github.com/openshift/client-go/config/clientset/versioned"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// OpenShift is an Client struct which will be used for all OpenShift related functions to interact with the existing
+// Cluster.
+type OpenShift struct {
+	Client *clientset.Clientset
+}
+
+// GetOpenShift creates client for the current OpenShift cluster. If Kubeconfig is provided, it is used to create client,
+// otherwise it uses in-cluster config.
+func GetOpenShift(kubeConfigPath string) (*OpenShift, error) {
+	//log.Printf("kubeconfig source: %s", kubeConfigPath)
+	var rc *rest.Config
+	var err error
+
+	if kubeConfigPath == "" {
+		// InClusterConfig uses default service account or service account provided by the pod to obtain config.
+		rc, err = rest.InClusterConfig()
+	} else {
+		rc, err = clientcmd.BuildConfigFromFlags("", kubeConfigPath)
+	}
+	if err != nil {
+		return nil, fmt.Errorf("error creating the config object %v", err)
+	}
```

Comment:
It seems like you are reimplementing something that is built into the controller-runtime already, please use
https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/client/config/config.go#L64
instead if possible